lib/compat/wordpress-6.1/class-gutenberg-rest-edit-site-export-controller.php

```diff
@@ -0,0 +1,189 @@
+ * and template parts either as a zip bundle or as edits to the current enabled
+ * theme's files.
+ *
+ * @since 5.9.0
+ *
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Edit_Site_Export_Controller_6_1 extends Gutenberg_REST_Edit_Site_Export_Controller {
+
+
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/export_to_theme_files',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'export_to_theme_files' ),
+					'permission_callback' => array( $this, 'permissions_check' ),
+				),
+			),
+			true
+		);
+
+		parent::register_routes();
+	}
+
+	/**
+	 * Update the files of the currently enabled block theme with the template and
+	 * template part customisations and additions made in the site editor.
+	 *
+	 * @return WP_Error|void
+	 */
+	public function export_to_theme_files() {
+		try {
+			$this->update_template_and_parts_files();
+			$this->update_theme_json_file();
+		} catch ( Exception $e ) {
+			return new WP_Error( 'rest_cant_update_theme_files', __( 'Error updating theme files. ', 'gutenberg' ) . $e->getMessage(), array( 'status' => 500 ) );
+		}
+
+		$this->clear_user_customizations();
+		return rest_ensure_response(  array( "theme_files_updated" => true ) );
+
+
+	}
+
+	protected function update_template_and_parts_files() {
+
+		$user_customisations = $this->get_user_customisations();
+
+		foreach ( $user_customisations['templates'] as $template ) {
+			$fileToUpdate = get_template_directory() . '/templates/' .
+				$template->slug . '.html';
+			if ( ! file_exists( $fileToUpdate ) ) {
+				throw new Exception("Theme file does not exist: " . $fileToUpdate);
+			}
+			if ( ! wp_is_writable( $fileToUpdate ) ) {
+				throw new Exception("Theme file is not writeable: " . $fileToUpdate);
+			}
+			file_put_contents(
+				$fileToUpdate,
+				$template->content
+			);
+		}
+
+		foreach ( $user_customisations['template_parts'] as $template_part ) {
+			$fileToUpdate = get_template_directory() . '/parts/' .
+				$template_part->slug . '.html';
+			if ( ! file_exists( $fileToUpdate ) ) {
+				throw new Exception("Theme file does not exist: " . $fileToUpdate);
+			}
+			if ( ! wp_is_writable( $fileToUpdate ) ) {
+				throw new Exception("Theme file is not writeable: " . $fileToUpdate);
+			}
+			file_put_contents(
+				$fileToUpdate,
+				$template_part->content
+			);
+		}
+	}
+
+	protected function update_theme_json_file() {
+		$fileToUpdate = get_template_directory() . '/theme.json';
+		if ( ! file_exists( $fileToUpdate ) ) {
+			throw new Exception("Theme file does not exist: " . $fileToUpdate);
+		}
+		if ( ! wp_is_writable( $fileToUpdate ) ) {
+			throw new Exception("Theme file is not writeable: " . $fileToUpdate);
+		}
+		file_put_contents(
+			$fileToUpdate,
+			gutenberg_export_theme_json()
+		);
+	}
+
+	protected function get_user_customisations() {
```

Comment:
Should this be called something like `get_user_customisations_and_reset`?

Reply:
No, because it does not reset anything., does it?